malaria-ios/Managers/AchievementManager.swift

```diff
@@ -0,0 +1,174 @@
+import Foundation
+
+/// Handles general achievement logic throughout the app.
+
+class AchievementManager {
+  static let sharedInstance = AchievementManager()
+  
+  var context: NSManagedObjectContext! =
+    CoreDataHelper.sharedInstance.createBackgroundContext()!
+  
+  private var achievements: [Achievement] {
+    let fetchRequest = NSFetchRequest(entityName: "Achievement")
+    
+    do {
+      let result = try context.executeFetchRequest(fetchRequest) as! [Achievement]
+      return result as [Achievement]
+    } catch let error as NSError {
+      Logger.Error("Could not fetch \(error), \(error.userInfo)")
+    }
+    return []
+  }
+  
+  /**
+   Each achievement has it properties that needs to be fulfilled
+   in order to unlock.
+   */
+  
+  private var properties: [AchievementProperty] {
+    let fetchRequest = NSFetchRequest(entityName: "AchievementProperty")
+    
+    do {
+      let result = try context.executeFetchRequest(fetchRequest) as! [AchievementProperty]
```

Comment:
I think you can return directlyy the AchivementProperty.Retrieve(AchivementProperty.self)